### ISSUE TITLE
Refactor all remaining config options to be checked at runtime

### ISSUE THREAD

**Issue by EliAndrewC** [magfest/ubersystem#1103]
One thing we did with the recent refactor was to make it so that we check most config options at runtime rather than at import time.  This is important because it means that if/when we implement a way to change config without restarting the process, the site will react properly instead of still doing whatever was decided when the process first started.

There are only a few places where we need to make changes.  Off the top of my head:
* We're passing a lot of constants to ``Choice`` and ``MultiChoice`` columns in ``uber.models``, so those classes will need a minor refactor allowing them to accept functions in addition to lists.  (There are theoretically other ways we could handle that, but this seems like the simplest one, which won't require much work.)
* The code for Season Pass events only checks the config at import time.  We should probably just change the background task to iterate over all events instead of having a separate one declared for each.  Or something along those lines.
* The code for the Department Checklist only checks the config at import time.  We could just make the same change we make for Season Pass events and have the background task read the config at runtime, or something morally equivalent.

There are a few more places where we reference ``c`` config options at the global level, but I don't think  any of those will need to be mutable.  For example, when we say

```python
@all_renderable(c.PEOPLE)
class Root:
    ...
```

then even if we edit the textual description of the ``c.PEOPLE`` integer constant, nothing would actually change in terms of how that's being checked and used.  So I think the above three things might literally be the only holdouts.

**Comment by EliAndrewC** [comment]
As soon as I posted this, I realized that there are a few things which would be more trouble than it's worth to change, and which we wouldn't really want to change anyway.  They're basically just:
* our database connection string, which contains the password we use to connect to the db
* our CherryPy web server config, e.g. the port we listen on

That's pretty much it though.  We probably don't need special cases for those, but if people feel we do then theoretically we could move them to a separate object or something.  I think that would be more trouble than it's worth, personally.

**Comment by thaeli** [comment]
Agreed that we probably don't need to worry about mutability in stuff like `@all_renderable(c.PEOPLE)` *as part of this refactor* - we very much need to revisit our permissions setup separately though, and I expect all of those global uses of `c` variables will go away once we refactor our permissions.

I think it would be better programming practice, and safer, to move items such as the db connection string to a separate, 'local configuration' object. That way, we don't have to worry about the remote synchronization changing that config, and it's very clear to any future programmers that this is a special case of "configuration that absolutely will not change during runtime". Otherwise, we end up with "changes to the `c` object are applied immediately, except when they're not, and some of those not-applied-immediately items need to use a separate backing store for security reasons" and that's confusing - I'd even call it a sort of technical debt.
